torch/csrc/jit/passes/remove_redundant_profiles.cpp

```diff
@@ -0,0 +1,46 @@
+#include <torch/csrc/jit/passes/dead_code_elimination.h>
+
+#include <torch/csrc/jit/ir/alias_analysis.h>
+#include <torch/csrc/jit/ir/ir_views.h>
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/utils/memory.h>
+
+namespace torch {
+namespace jit {
+
+void RemoveRedundantProfiles(Block* block, AliasDb& db) {
+  for (auto it = block->nodes().end()->reverseIterator();
+       it != block->nodes().begin();) {
+    Node* n = *it;
+    it++;
+
+    for (Block* b : n->blocks()) {
+      RemoveRedundantProfiles(b, db);
+    }
+
+    if (n->kind() != prim::profile ||
```

Comment:
Should we do the same for `profile_optional`?

Reply:
`profile_optional` profiles the definition instead of each use (bc None type is immutable unlike tensor type), so it's not possible that there's a redundant profile_optional